### PR TITLE
Revisit slug configuration

### PR DESCRIPTION
**Pull request by levinkeo**
Continuation of the work done by @anthmatic in #6 in the context of where we landed for https://github.com/HubSpot/cms-react-boilerplate
- Adds multiple instance support for both modules
- updates `{{ module }}` prior to sending to app to pre-evaluate any module data that requires extra HubL (like `content_by_id()`, `slug` -- I'm on the fence about this approach and could use some feedback)
- re-organizes the `module|tojson` pass in events app to remain consistent with the CMS React boilerplate

TODO:
- [x] Should `my-events` be made configurable from the events app module, too?

### CODE REVIEW

**Comment by anthmatic** [comment]
> Should my-events be made configurable from the events app module, too?

I think it would be interesting to see what this looks like with a required page selector field. Additionally, I think it would be nice to have the `my-events` route nested under the main app slug (`/events/my-events`)

**Comment by levinkeo** [comment]
> I think it would be interesting to see what this looks like with a required page selector field. Additionally, I think it would be nice to have the `my-events` route nested under the main app slug (`/events/my-events`)

@anthmatic Ah, yeah changing to `<events>/<my-events>` isn't something I'd considered-- I like that much better. 

By required page selector, do you mean a required page selector in both modules and for which pages? `events` page selector in the upcoming events module and `my-events` in the main event app? 

For the events app to have a concept of the app "root url", relying on a page selector field worked more consistently for me in testing than using `content.slug`, but I need to dig into that a bit more today. I think having to select the page you're currently editing is a weird flow, so getting `slug` to work would be preferable.

**Comment by anthmatic** [comment]
I was thinking the main event listing module would have 2 page fields:
- events page
- my events page

But I agree, if `content.slug` is reliable, that would be nicer, for sure.

I don't think the upcoming events module would need to know about the my events page.

File: src/App.js [hunk 1]
```diff
@@ -30,13 +32,13 @@ function App() {
...
             {state.contact.isLoggedIn ? (
-              <Link to="/my-events" className="event-button">
+              <Link to={myEventsPath} className="event-button">
                 View my Events
               </Link>
             ) : (
               <a
                 style={{ textDecoration: 'none' }}
-                href="/_hcms/mem/login?redirect_url=/my-events"
+                href={`/_hcms/mem/login?redirect_url=/${my_events_page}`}
```
Comment (anthmatic) [inline src/App.js:41]: should this be an absolute url?

Comment (levinkeo) [reply src/App.js:41]: Hmm, I guess it probably should be relative since the rest of the app relies on relative paths.

**Comment by levinkeo** [comment]
One big downside to this configurability is the necessity of selecting _both_ the `my events` and `events` pages in every instance of the Events app module. This adds to the setup time and increases the number of HS screens you're flipping between

**Comment by achong07** [comment]
Thanks for clarifying @levinkeo! That makes sense. I don't have any objections to this at the moment, but agreed it'll be good to see in action, if possible!

**Pull request merged**